### PR TITLE
Traffic Control API

### PR DESCRIPTION
**Pull request by arrubin**

### CODE REVIEW

File: traffic_control/api.txt [hunk 1]
```diff
@@ -0,0 +1,29 @@
+passenger list                GET    /passengers
+passenger details             GET    /passengers/:id
+passenger itineraries         GET    /passengers/:id/itineraries
+passenger itinerary details   GET    /passengers/:id/itineraries/:id
+new passenger                 POST   /passengers
+new passenger itinerary       POST   /passengers/:id/itineraries
```
Comment (jmmastey) [inline traffic_control/api.txt]: Itineraries probably need an update via `PATCH` or `PUT`

File: traffic_control/api.txt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+airplane details              GET    /airplanes/:id
+airplane segments             GET    /airplanes/:id/segments
+airplane refuelings           GET    /airplanes/:id/refuelings
+new airplane                  POST   /airplanes
+new airplane segment          PUT    /airplanes/:id:/segments
+delete airplane segment       DELETE /airplanes/:id/segments/:id
+
+airport list                  GET    /airports
+airport details               GET    /airports/:id
+airport flights               GET    /airports/:id/flights
```
Comment (jmmastey) [inline traffic_control/api.txt]: I'm not positive whether this overlaps the `GET /flights` search path.

File: traffic_control/api.txt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+airport airplanes             GET    /airports/:id/airplanes
```
Comment (jmmastey) [inline traffic_control/api.txt]: Do airports correspond to airplanes? Aside from the possible duplication from `GET /airplanes/:id`, I'm not sure this models the domain well.

File: traffic_control/api.txt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+new airport                   POST   /airports
```
Comment (jmmastey) [inline traffic_control/api.txt]: No updates allowed to airports?

**Comment by jmmastey** [comment]
These routes are good, left a few notes. Nice work.

File: traffic_control/api.txt (hunk 3 above)
Comment (arrubin) [inline traffic_control/api.txt]: If this is for traffic control, I figured that knowing which airplanes are at a particular airport might be useful.

File: traffic_control/api.txt (hunk 2 above)
Comment (arrubin) [inline traffic_control/api.txt]: The idea was to get flights that involved a particular airplane.

**Pull request closed without merging**